p2p/protocol/identify/obsaddr.go

```diff
@@ -46,13 +51,36 @@ type ObservedAddr struct {
 	LastSeen time.Time
 }
 
-func (oa *ObservedAddr) activated(ttl time.Duration) bool {
+func (oa *ObservedAddr) activated() bool {
+
 	// We only activate if other peers observed the same address
 	// of ours at least 4 times. SeenBy peers are removed by GC if
 	// they say the address more than ttl*ActivationThresh
 	return len(oa.SeenBy) >= ActivationThresh
 }
 
+func (oa *ObservedAddr) numInbound() int {
+	count := 0
+	for obs := range oa.SeenBy {
+		if oa.SeenBy[obs].connDirection == network.DirInbound {
+			count++
+		}
+	}
+
+	return count
+}
+
+func (oa *ObservedAddr) GroupKey() string {
+	key := ""
+	protos := oa.Addr.Protocols()
+
+	for i := range protos {
+		key = key + protos[i].Name
```

Comment:
This is fine but kind of sloppy. We should either use a separator (`/ip4/udp/quic`) or concatenate `protocol.VCode`s (these are prefix-free codes so we don't need separators).

Reply:
Using a separator now.